features/192-asset-gear/index.md

```diff
@@ -0,0 +1,671 @@
+- `headers.content-md5`: The [Content-MD5](https://tools.ietf.org/html/rfc1864) header. Depending on the upload method, it may be ignored.
+- `headers.cache-control`: The Cache-Control header. If it is not given, a default one is added. To remove the header, set it to an empty string.
+- `headers.access-control-*`: The CORS headers. They are only useful if the developer needs to retrieve the asset with Fetch API.
+
+##### Request Example
+
+```json
+{
+  "headers": {
+    "content-type": "image/png"
+  }
+}
+```
+
+#### Response JSON Schema
+
+```json
+{
+  "type": "object",
+  "properties": {
+    "asset_id": { "type": "string" },
+    "url": { "type": "string" },
+    "method": { "type": "string" },
+    "headers": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "name": { "type": "string" },
+          "value": { "type": "string" }
+        },
+        "required": ["name", "value"]
+      }
+    },
+    "form": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "name": { "type": "string" },
+          "value": { "type": "string" },
+          "filename": { "type": "string" }
+        },
+        "required": ["name"]
+      }
+    }
+  },
+  "required": ["asset_id", "url", "method", "headers"]
+}
+```
+
+- `asset_id`: The asset ID. It becomes valid when the asset is uploaded.
+- `url`: The presigned URL to upload the asset.
+- `method`: The method to be used in the upload request.
+- `headers`: The headers to be included in the upload request.
+- `form`: The form data to be included in the upload request.
+  - `filename`: If the form name-value pair has this property, it is the field for the asset.
+
+##### Response Example
+
+```json
+{
+  "asset_id": "myimage.png",
+  "url": "https://storage.skygearapps.com/bucket/myappid/myimage.png?...",
+  "method": "POST",
+  "headers": [
+    { "name": "content-type", "value": "multipart/form-data" }
+  ],
+  "form": [
+    { "name": "file", "filename": "myimage.png" },
+    { "name": "Policy", "value": "..." }
+  ]
+}
+```
+
+#### Server Specification
+
+1. Let `ext` be an empty string.
+1. If `headers.content-type` is present, set `ext` to the file extension derived from it.
+1. If `exact_name` is given and non-empty, let `name` be `exact_name`. Set `headers.cache-control` to `no-cache` if it is absent.
+1. Otherwise let `name` to be the concatenation of `prefix`, a random string and `ext`. Set `headers.cache-control` to `max-age: 3600` if it is absent.
+1. Let `asset_id` be `/<app-id>/<name>`.
+1. Remove any header in `headers` whose value is empty string.
+1. Presign `asset_id` with `headers` and `access`.
+1. Return the presigned request.
+
+#### Client Specification
+
+If the `content-type` header in the presign request is `multipart/form-data`, the client must submit a multipart form request.
```

Comment:
According to offline discussion with @louischan-oursky , you can still found `multipart/form-data` tutorial as top result when you google on how to upload file in react/angular. Please make sure the `FormData` is working with the upload API.

Reply:
I am going to add a endpoint to accept `multipart/form-data` with a single file.